### ISSUE TITLE
feature: include line numbers in filenames when matched

### ISSUE THREAD

**Issue by sr105** [facebook/PathPicker#154]
when matching the output from grep, fpp highlights the filename including the trailing :LINE_NUMBER. It would be nice to have an option to save the line number in the output. I take the output of fpp and send it to emacs through a script. I'd like to parse out the line number and use it.

**Comment by sr105** [comment]
Dug through the code and figured out that line numbers are supported for "emacs", "joe", and "vim". So, I created a symlink to my editor script and named it joe. Then, I set FPP_EDITOR=joe. That didn't work because fpp appears to follow the symlink. So, I named my script joe and created a symlink with the old name and all is good. 

It would still be nice to support outputting file:line as an option and let the user's command/editor parse it.

**Comment by pcottle** [comment]
Get @sr105, thanks for the file. Yeah the main issue here is that the support and formatting for various editors can vary quite widely -- for instance vim only supports setting the line number when its only opening //one// file, whereas emacs and joe are quite different.

Out of curiosity what is your `$EDITOR` set to? That's the variable we look at (if FPP_EDITOR is not set).

This is a similar issue for the command mode -- we have `$F` as a token to represent all files joined by a space, but if the user wanted line numbers, its not clear what format they would want them in.

Just trying to understand your specific usecase -- id rather make a slightly more magical but frictionless solution for a specific editor than introduce too much complexity

**Comment by pcottle** [comment]
Gonna close this out since it seems you're unblocked but lets continue the discussion

**Comment by sr105** [comment]
The EDITOR and symlink stuff must have been an error on my part because I can't reproduce it.

For now, I have a symlink named joe that points to a shell script. The script takes the list of filenames and sends them over ssh to another script which uses emacsclient to open them on another machine. I mangle each path like this before I send them:

    for each path in "$@":
         path = /ssh:user@host:$(realpath $path)

I was thinking that when the EDITOR is not explicitly emacs, joe, or vim, there could be an option (off by default like today) to pass the paths in $F as path:line and let the user parse them. The option could take an optional argument to specify style:

    -X [type]
    where type is:
        v - vi style
        e - emacs style
        g - grep style, file:line (default)
    if EDITOR or FPP_EDITOR is vi, emacs, or joe, -X is ignored and line number styles are set automatically

**Comment by pcottle** [comment]
Ah cool, glad you got unblocked! I think having multiple formatting options is a bit intense (since the emacs and vim style is already handled) but I wouldn't be opposed to a PR that adds an option to output **all** the files and line numbers separated by some kind of token.

Also keep in mind that sometimes we don't have filenumbers available (for certain types of parsing), so you might get 0's in place of an actual filenumber.

Anyways have fun path-picking!

**Comment by sr105** [comment]
-X [char]      output file and line number for each file separated by char. Default char is ':'. If no line number is found, 0 is used. Example: file.txt:78.

**Comment by sr105** [comment]
I think you're right about limiting the options. If you can parse one format, you can parse any.
